File: app/src/main/java/me/gabeg/sicksends/SsBoulderViewModel.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Boulder view model.
 */
public class SsBoulderViewModel
	extends SsProblemViewModel<SsBoulderRepository, SsBoulder>
{

	/**
	 */
	public SsBoulderViewModel(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemViewModel#buildRepository(Application)
	 */
	public void buildRepository(Application app)
	{
		this.mRepository = new SsBoulderRepository(app);
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/boulder/SsBoulderViewModel.kt
package me.gabeg.sicksends.db.boulder

import androidx.lifecycle.SavedStateHandle
import dagger.hilt.android.lifecycle.HiltViewModel
import me.gabeg.sicksends.db.generic.SsGenericProblemViewModel
import javax.inject.Inject

/**
 * Boulder view model.
 */
@HiltViewModel
class SsBoulderViewModel @Inject constructor(
	savedStateHandle: SavedStateHandle,
	repo : SsBoulderRepository)
		: SsGenericProblemViewModel<SsBoulderProblem>(savedStateHandle, repo)
